crypto/src/builder.rs

```diff
@@ -13,18 +15,30 @@ use crate::{
     Fr, Note, Output, Spend, Value,
 };
 
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Chain ID not set")]
+    NoChainID,
+    #[error("Expiry height not set")]
+    ExpiryHeightNotSet,
```

Comment:
The expiry height should probably actually be optional, it's fine for people not to set it.